### PR TITLE
Allow assets to be defined in articles.json

### LINKED ISSUE #925: Asset paths defined in articles.json not converted
Use case: I have an extension which requires custom assets to be defined. When the course is built, these are copied into the correct place, but the filepath in the JSON is incorrect.

Only course, components and content objects work:
https://github.com/adaptlearning/adapt_authoring/blob/master/lib/outputmanager.js#L656-L657

### PR DESCRIPTION
**Pull request by taylortom**
Fixes #925

### CODE REVIEW

File: lib/outputmanager.js [hunk 1]
```diff
@@ -653,14 +654,21 @@ OutputPlugin.prototype.writeCourseAssets = function(tenantId, courseId, destinat
                   // TODO -- This global replace is intended as a temporary solution
                   var replaceRegex = new RegExp("course/assets/" + asset.filename, 'gi');
 
-                  jsonObject['component'] = JSON.parse(JSON.stringify(jsonObject['component']).replace(replaceRegex, "course/" + jsonObject['config']._defaultLanguage + "/assets/" + encodeURIComponent(asset.filename)));
-                  jsonObject['course'] = JSON.parse(JSON.stringify(jsonObject['course']).replace(replaceRegex, "course/" + jsonObject['config']._defaultLanguage + "/assets/" + encodeURIComponent(asset.filename)));
+                  var lang = jsonObject['config']._defaultLanguage;
+                  var newAssetPath = "course/" + lang + "/assets/" + encodeURIComponent(asset.filename);
+
+                  var objects = ['config', 'course', 'article', 'block', 'component'];
```
Comment (taylortom) [inline lib/outputmanager.js]: Set up to do the replace for:
- config
- course
- article
- block
- component

Comment (brian-learningpool) [inline lib/outputmanager.js]: I don't think there was a need to declare the array of `objects` like this.  There is already a constant you can use: `Object.keys(Constants.CourseCollections)`

**Comment by brian-learningpool** [comment]
+1

File: lib/outputmanager.js [hunk 2]
```diff
@@ -229,9 +229,10 @@ OutputPlugin.prototype.writeCourseJSON = function(jsonObject, destinationFolder,
 
     async.each(Object.keys(Constants.CourseCollections), function(collectionType, callback) {
       var data = JSON.stringify(outputJson[collectionType], undefined, 2);
-      var filename = (collectionType === 'config') 
+      var filename = (collectionType === 'config')
         ? path.join(destinationFolder, Constants.CourseCollections[collectionType].filename)
         : path.join(destinationFolder, outputJson['config']._defaultLanguage, Constants.CourseCollections[collectionType].filename);
+        debugger;
```
Comment (dennis-learningpool) [inline lib/outputmanager.js]: whoops

Comment (taylortom) [inline lib/outputmanager.js]: :sweat_smile:

**Comment by dennis-learningpool** [comment]
+1

**Pull request merged**